transactions/idTableStaking/admin/add_approved_and_limits.cdc

```diff
@@ -0,0 +1,54 @@
+import FlowIDTableStaking from 0xIDENTITYTABLEADDRESS
+
+/// This transaction adds node IDs to the list of approved nodes in
+/// the ID table. 
+/// It also increases the candidate node limit and slot limits
+/// by the number of nodes who are added
+///
+/// If any of the provided nodes already exist in the ID table, this
+/// transaction will not revert (idempotent)
+
+transaction(ids: [String]) {
+
+    // Local variable for a reference to the ID Table Admin object
+    let adminRef: &FlowIDTableStaking.Admin
+
+    prepare(acct: AuthAccount) {
+        // borrow a reference to the admin object
+        self.adminRef = acct.borrow<&FlowIDTableStaking.Admin>(from: FlowIDTableStaking.StakingAdminStoragePath)
+            ?? panic("Could not borrow reference to staking admin")
+    }
+
+    execute {
+		let nodeIDs = FlowIDTableStaking.getApprovedList()
+
+		let candidateNodeLimits = FlowIDTableStaking.getCandidateNodeLimits()
```

Comment:
We had decided to set the candidate limits to zero for non-access because we didn't want any nodes to register unless we already approved them, but I just realized that that would prevent any new nodes from registering until we increase the limit since we say that a node can't register unless there are open candidate slots.

I agree with your idea for when we have full staking auctions, but for now they've said to just keep the candidate slots closed. I think we still need at least one for each role though, but maybe this is a good enough reason to allow a lot of candidate slots for non-access nodes since spamming is disincentivized anyway by the minimum stake.

I'll also set those restrictions to the set limits methods

Reply:
actually, I can't add those restrictions yet, because with what leadership has said about having minimal candidate slots, we can't have those restrictions because the staking slot limit would basically always be higher for non-access nodes. I'll create another issue to discuss it though

Reply:
We have more than a week before the mainnet upgrade, so we'll have time to discuss it

Reply:
I talked to vishal and he is okay with setting the candidate limits to higher values since spam is disincentivized by the minimum stake requirement. I also just created an issue (#341) for the limits pre-conditions and we'll include that in the next upgrade